File: Backend/api/authRoutes.js
const express = require('express');
const AuthController = require('../controllers/authController');
const auth = require('../middlewares/auth');

const router = express.Router();
router.post('/register', AuthController.register);
router.post('/login', AuthController.login);
router.get('/profile', auth, AuthController.getProfile);
router.post('/logout', auth, AuthController.logout);

module.exports = router;

File: backend_python/api/auth_routes.py
from flask import Blueprint

from controllers.auth_controller import AuthController
from middlewares.auth import auth_required

bp = Blueprint("auth", __name__)

bp.add_url_rule("/register", view_func=AuthController.register, methods=["POST"])
bp.add_url_rule("/login", view_func=AuthController.login, methods=["POST"])
bp.add_url_rule("/profile", view_func=auth_required(AuthController.get_profile), methods=["GET"])
bp.add_url_rule("/logout", view_func=auth_required(AuthController.logout), methods=["POST"])
